care/emr/api/viewsets/scheduling/booking.py

```diff
@@ -49,20 +51,35 @@ class TokenBookingFilters(FilterSet):
     status = CharFilter(field_name="status")
     date = DateFromToRangeFilter(field_name="token_slot__start_datetime__date")
     slot = UUIDFilter(field_name="token_slot__external_id")
-    user = UUIDFilter(method="filter_by_user")
+    user = CharFilter(method="filter_by_users")
     patient = UUIDFilter(field_name="patient__external_id")
 
-    def filter_by_user(self, queryset, name, value):
+    def filter_by_users(self, queryset, name, value):
         facility_external_id = self.request.parser_context.get("kwargs", {}).get(
             "facility_external_id"
         )
-        resource = SchedulableUserResource.objects.filter(
-            user__external_id=value,
-            facility__external_id=facility_external_id,
-        ).first()
-        if not resource:
+        user_external_ids = value.split(",") if value else []
+
+        token_slots = TokenSlot.objects.filter(
+            resource_id__in=Subquery(
+                SchedulableUserResource.objects.filter(
+                    user_id__in=Subquery(
+                        User.objects.filter(external_id__in=user_external_ids).values(
+                            "id"
+                        )
+                    ),
+                    facility_id__in=Subquery(
```

Comment:
This will always be a single object right ? can we fetch and filter instead of subquery, we will switch to a cached approach to get facility objects like user objects soon enough